### PR TITLE
Restart with Tests

### PR DESCRIPTION
**Pull request by philippweikert**

### CODE REVIEW

File: backend/src/main/java/de/neuefische/capstone/plantcomponents/PlantController.java [hunk 1]
```diff
@@ -15,12 +17,33 @@ public class PlantController {
...
+    public Collection<Plant> getAllPlants(Principal principal) {
+        return plantService.getAllPlantsByUser(principal.getName());
     }
+
+    @GetMapping("/{id}")
+    public ResponseEntity <Plant> getPlantById(@PathVariable String id, Principal principal){
+        return ResponseEntity.of(plantService.getPlantByIdAndUser(id, principal.getName()));
+    }
+
+    @GetMapping("{/searchedPlant}")
```
Comment (gossie) [inline backend/src/main/java/de/neuefische/capstone/plantcomponents/PlantController.java:34]: That won't work, it needs to be like this: `"/{searchedPlant}"`

File: backend/src/main/java/de/neuefische/capstone/plantcomponents/PlantService.java [hunk 2]
```diff
@@ -3,21 +3,46 @@
...
+
+    public List<Plant> getMatchingPlant(String searchedPlant, String username) {
+        return plantRepo.findAllByUser(username).stream()
+                .filter(plant -> plant.getNonScName().toLowerCase().contains(searchedPlant.toLowerCase()))
+                .toList();
+    }
+
+    public Plant deletePlant(String id, String username) {
+        Optional<Plant> optionalPlant = plantRepo.findByIdAndUser(id, username);
+        Plant plant = optionalPlant.orElseThrow(() -> new IllegalArgumentException("No plant with ID: " + id + " found!"));
```
Comment (gossie) [inline backend/src/main/java/de/neuefische/capstone/plantcomponents/PlantService.java:32]: You could return an empty `Optional` if no plant was found and use `ResponseEntity.of` in the controller to return a 404 if no plant was found.

File: backend/src/main/java/de/neuefische/capstone/plantcomponents/PlantService.java [hunk 3]
```diff
@@ -3,21 +3,46 @@
...
+        plantRepo.delete(plant);
+        return plant;
+    }
+
+    public Optional<Plant> getPlantByIdAndUser(String id, String username) {
+        return plantRepo.findByIdAndUser(id, username);
+    }
+
+    public Plant changePlant (Plant changedPlant){
```
Comment (gossie) [inline backend/src/main/java/de/neuefische/capstone/plantcomponents/PlantService.java:41]: Same here

**Review by gossie — CHANGES_REQUESTED** [review]

File: backend/src/main/java/de/neuefische/capstone/plantcomponents/PlantController.java (hunk 1 above)
Comment (philippweikert) [reply backend/src/main/java/de/neuefische/capstone/plantcomponents/PlantController.java:34]: Sorry, Typo Phd at work!

**Review by gossie — APPROVED** [review]

**Pull request merged**